### DISCUSSION TITLE
Release piler as add-on for Nextcloud or Home Asssistant

### DISCUSSION THREAD

**Discussion by axlFThn** [jsuto/piler#188] [Ideas]
Nextcloud and Home Assistant are popular tools for the self-host community, making it possible to host archives in your own network. Nextcloud in particular, I see as one of the leading cloud-solutions in the open-source community. 

However, nextcloud lacks the possibility to archive mail in a proper way. Here, I see piler as a great solution to this missing feature. Would it be possible to release piler as an add-on for either nextcloud or homeassistant? Seems like this could be easily done with docker?

Elaboration:
In [this nextcloud thread on github](https://github.com/nextcloud/mail/issues/9207), a possible integration of piler was discussed. However, I think the purpose and potential of such an integration was not fully understood. Of course, when using nextcloud in an docker environment, piler can already be easily attached in an adjacent docker, I assume. However, in systems where nextcloud is deployed in different ways, this becomes more difficult. Turning piler into a nextcloud-integration, piler would be able to reach a large community of users/businesses which potentially need to archive emails, alongside usual archiving solutions. 

For me, its more about having a place to store a large amount of emails which I like to keep in the long-run. Using home-assistant as the base OS, I cannot deploy other docker images without tainting the system and loosing automatic updates and system security. Hence, an hassio-addon from the existing docker-image would be the shortest way for me.

Best regards,
Alexander

**Comment by jsuto** [comment]
I commented in the linked nextcloud issue. Let's see what they will respond.
